### PR TITLE
Fix: Application name update checks

### LINKED ISSUE #1581: [Bug] Space should not be accepted as First character for Application name
Steps to reproduce
1) user creates a new application 
2) User is navigated inside the application
3) Can edit the name of the application
4) Add Space and navigate to Home page 
and observe

Observation
It is observed that the user is not displayed any name in the home page 

Expectation 
User should be allowed to provide only Space as the first character 
Screenshot 2020-11-05 at 1 35 05 PM

### LINKED ISSUE #1582: [Bug] Appropriate information must be displayed to user on updating application name
Steps to reproduce
1) Navigate inside the application
2) Click on the application name to edit 
3) Add a name 
4) Click outside the page 
and observe 

Observation:
It is observed that the  user is displayed with an information message stating "Application updated"

Expectation:
User must be displayed with message stating "Application name is updated" making it more specific of what is updated in the application

### LINKED ISSUE #1584: [Bug] On empty application name user must be displayed with an error message inside the application
Steps to reproduce :
1) Navigate inside the application
2) Click on the application name 
3) Delete the entire name of the application
4) Click outside the header 
and observe

Observation 
It is observed that the last word or the name cleared stays as the application name 

Expectation
When user tries to save an empty field as application name user must be displayed with an error message and work similar to the behaviour as on homepage

### PR DESCRIPTION
**Pull request by devrk96**
Fixes #1581 
Fixes #1582
Fixes #1584

### CODE REVIEW

**Comment by mohanarpit** [comment]
@devrk96 Is it possible to write Jest/Cypress tests to assert these bug fixes?

File: app/client/src/components/ads/EditableText.tsx [hunk 1]
```diff
@@ -173,14 +174,29 @@ export const EditableText = (props: EditableTextProps) => {
...
+        AppToaster.show({
+          message: "Application name can't be empty",
+          type: "error",
+        });
+      } else {
+        AppToaster.show({
+          message: "Application name updated",
+          type: "success",
+        });
+      }
```
Comment (hetunandu) [inline app/client/src/components/ads/EditableText.tsx:188]: Application name specific code here in the design system file. You need to extract this validation somewhere else

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Review by hetunandu — APPROVED** [review]

File: app/client/src/components/ads/EditableTextWrapper.tsx [hunk 2]
```diff
@@ -96,6 +102,12 @@ export default function EditableTextWrapper(props: EditableTextWrapperProps) {
           if (props.isInvalid) {
             setIsValid(Boolean(props.isInvalid(value)));
             return props.isInvalid(value);
+          } else if (value === "") {
```
Comment (hetunandu) [inline app/client/src/components/ads/EditableTextWrapper.tsx:105]: Trim the name before checking for this

**Review by hetunandu — CHANGES_REQUESTED** [review]
Minor change needed

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**